jaeger-core/src/main/java/com/uber/jaeger/Configuration.java

```diff
@@ -385,6 +419,201 @@ public Integer getMaxQueueSize() {
+     * @param authUsername the HTTP Basic Auth username
+     * @param authPassword the HTTP Basic Auth password
+     */
+    public SenderConfiguration(String endpoint, String authUsername, String authPassword) {
+      this(null, null, endpoint, null, authUsername, authPassword);
+    }
+
+    private SenderConfiguration(String agentHost, Integer agentPort, // agent-related properties
+                                String endpoint, // only required property when using the collector
+                                String authToken, // one valid auth combination
+                                String authUsername, String authPassword) { // the other valid auth
+      this.agentHost = agentHost;
+      this.agentPort = agentPort;
+      this.endpoint = endpoint;
+      this.authToken = authToken;
+      this.authUsername = authUsername;
+      this.authPassword = authPassword;
+    }
+
+    /**
+     * Returns a sender if one was given when creating the configuration, or attempts to create a sender based on the
+     * configuration's state.
+     * @return the sender passed via the constructor or a properly configured sender
+     */
+    public Sender getSender() {
+      // if we have a sender, that's the one we return
+      if (null != this.sender) {
+        return this.sender;
+      }
+
+      if (null != endpoint && !endpoint.isEmpty()) {
+        log.debug("The endpoint is set. Attempting to configure HttpSender for it.");
+        Interceptor authInterceptor = null;
+
+        if (null != authUsername && !authUsername.isEmpty()
+                && null != authPassword && !authPassword.isEmpty()) {
+          log.debug("Using HTTP Basic authentication with data from the environment variables.");
+          authInterceptor = getAddBasicAuthInterceptor(authUsername, authPassword);
+        } else if (null != authToken && !authToken.isEmpty()) {
+          log.debug("Auth Token environment variable found.");
+          authInterceptor = getAddTokenInterceptor(authToken);
+        }
+
+        // at this point, the authInterceptor can still be null
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+
+        if (null != authInterceptor) {
+          log.debug("All requests to the endpoint will have an Authentication header set.");
+          builder.addInterceptor(authInterceptor);
+        }
+
+        log.debug("Using the HTTP Sender to send spans directly to the endpoint.");
+        return new HttpSender(endpoint, builder.build());
+      }
+
+      log.debug("Using the UDP Sender to send spans to the agent.");
+      return new UdpSender(
+              stringOrDefault(this.agentHost, UdpSender.DEFAULT_AGENT_UDP_HOST),
+              numberOrDefault(this.agentPort, UdpSender.DEFAULT_AGENT_UDP_COMPACT_PORT).intValue(),
+              0 /* max packet size */);
+    }
+
+    public String getAgentHost() {
+      return agentHost;
+    }
+
+    public Integer getAgentPort() {
+      return agentPort;
+    }
```

Comment:
although I am not a big fan of lombok, but as long as we're using it in this repo, can't we just annotate the class?